lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Dictionary.java

```diff
@@ -1275,76 +1258,6 @@ boolean isDotICaseChangeDisallowed(char[] word) {
     return word[0] == 'İ' && !alternateCasing;
   }
 
-  private class EntryGrouper {
```

Comment:
The same-word-grouping logic in this class has been moved to `WordStorage.Builder`, and appended with building the reverse trie